src/SipHash.chpl

```diff
@@ -64,13 +64,40 @@ module SipHash {
   }
   
   proc sipHash64(msg: [] uint(8), D): uint(64) {
+    if contiguousIndices(msg) {
+      ref start = msg[D.low];
+      if D.high < D.low {
+        var res = computeSipHash(c_ptrTo(start), 0..#0, 8);
+        return res[1];
+      }
+      ref end = msg[D.high];
+      const startLocale = start.locale.id;
+      const endLocale = end.locale.id;
+      const hereLocale = here.id;
+      const l = D.size;
+      if startLocale == endLocale {
+        if startLocale == hereLocale {
+          var res = computeSipHash(c_ptrTo(start), 0..#l, 8);
+          return res[1];
+        } else {
+          var a = c_malloc(msg.eltType, l);
+          GET(a, startLocale, getAddr(start), l);
+          var h = computeSipHash(a, 0..#l, 8);
+          c_free(a);
+          return h[1];
+        }
+      }
+    }
```

Comment:
Nit, but instead of duplicating the logic, I'd probably add a function like:

```chpl
private proc computeSipHashLocalize(msg, D, param outlen: int)
```

or whatever that handles the c_ptr/localization logic and just call that from `sipHash64`/`sipHash128`

Reply:
good idea!